soweego/wikidata/api_requests.py

```diff
@@ -424,12 +427,107 @@ def _prepare_request(qids, props):
     return qid_buckets, request_params
 
 
+@lru_cache()
+def get_authenticated_session():
+    """
+    Returns the token to be used for authentication.
+    If token is not valid then a new one will be generated
+    """
+
+    wiki_api_dump_path = os.path.join(
+        constants.SHARED_FOLDER, constants.WIKIDATA_API_SESSION)
+
+    try:
+        with open(wiki_api_dump_path, 'rb') as f:
+            LOGGER.info('Previously authenticated session exists')
+            session = pickle.load(f)
+
+            # check if session is still valid
+            res = session.get(WIKIDATA_API_URL, params={
+                'action': 'query',
+                'assert': 'user',
+                'format': 'json'
+            })
+
+            # if the assert query failed then it means
+            # we need to renew the session
+            if 'error' in res.json().keys():
+                LOGGER.info('Session has expired and will be renewed')
+                raise AssertionError
+
+            return session
+
+    except (FileNotFoundError, AssertionError):
+        LOGGER.info('Obtaining new authenticated session')
```

Comment:
I suggest to refactor the following code: a few extracted functions would be enough.

Reply:
Good idea, login functionality has been refactored into smaller methods in 
a31d09f